### ISSUE TITLE
Snap report fails when configuring tests to be executed in parallel

### ISSUE THREAD

**Issue by dcordero** [fastlane/fastlane#14378]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Test counters are wrong when running Snap in a project, in which tests are configured to be executed in Parallel. (Edit Scheme -> Test -> Info tab -> Options -> Execute in parallel in Simulator)

![screenshot 2019-03-08 at 16 54 09](https://user-images.githubusercontent.com/2320840/54039384-16e44e80-41c3-11e9-98c6-679d87ef1556.jpg)

The final report looks like this 👇 :
```
+--------------------+-----+
|       Test Results       |
+--------------------+-----+
| Number of tests    | 0   |
| Number of failures | 0   |
+--------------------+-----+
```

On the other hand, running the same project with the same suite of tests, just changing the configuration to run the tests sequentially generates the following report:

```
+--------------------+-----+
|       Test Results       |
+--------------------+-----+
| Number of tests    | 959 |
| Number of failures | 0   |
+--------------------+-----+
```

##### Command executed

`bundle exec fastlane test`

Where the `test` lane is configured as follows:

```
fastlane_version '2.117.1'

require './helpers/code_coverage.rb'

desc 'Run all the tests suites of the project'
lane :test do
  clear_derived_data
  update_pods
  test_macos
  test_ios
  test_tvos
  post_to_slack(message: 'All tests successfully passed', success: true)
end

# Private Lanes

desc 'Run all the tests suites of the project for ios'
private_lane :test_ios do
  scan(scheme: 'ZattooFoundation-iOS')
  code_coverage(scheme: 'ZattooFoundation-iOS')
end

desc 'Run all the tests suites of the project for tvos'
private_lane :test_tvos do
  scan(scheme: 'ZattooFoundation-tvOS')
  code_coverage(scheme: 'ZattooFoundation-tvOS')
end

desc 'Run all the tests suites of the project for macos'
private_lane :test_macos do
  scan(scheme: 'ZattooFoundation-macOS')
  code_coverage(scheme: 'ZattooFoundation-macOS')
end
```

### Environment

<details>
  <pre> <details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                    |
| --------------------------- | -------------------------------------------------------- |
| OS                          | 10.14.3                                                  |
| Ruby                        | 2.4.1                                                    |
| Bundler?                    | true                                                     |
| Git                         | git version 2.17.2 (Apple Git-113)                       |
| Installation Source         | ~/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bin/fastlane |
| Host                        | Mac OS X 10.14.3 (18D109)                                |
| Ruby Lib Dir                | ~/.rbenv/versions/2.4.1/lib                              |
| OpenSSL Version             | OpenSSL 1.0.2n  7 Dec 2017                               |
| Is contained                | false                                                    |
| Is homebrew                 | false                                                    |
| Is installed via Fabric.app | false                                                    |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/              |
| Xcode Version               | 10.1                                                     |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.117.1 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin               | Version | Update-Status |
| -------------------- | ------- | ------------- |
| fastlane-plugin-ruby | 0.1.3   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| bundler                   | 1.16.5       |
| io-console                | 0.4.6        |
| CFPropertyList            | 2.3.6        |
| concurrent-ruby           | 1.1.4        |
| i18n                      | 0.9.5        |
| minitest                  | 5.11.3       |
| thread_safe               | 0.3.6        |
| tzinfo                    | 1.2.5        |
| activesupport             | 4.2.11       |
| public_suffix             | 2.0.5        |
| addressable               | 2.6.0        |
| ast                       | 2.4.0        |
| atomos                    | 0.1.3        |
| babosa                    | 1.0.2        |
| claide                    | 1.0.2        |
| clamp                     | 0.6.5        |
| fuzzy_match               | 2.0.4        |
| nap                       | 1.1.0        |
| cocoapods-core            | 1.5.3        |
| cocoapods-deintegrate     | 1.0.2        |
| cocoapods-downloader      | 1.2.2        |
| cocoapods-plugins         | 1.0.0        |
| cocoapods-search          | 1.0.0        |
| cocoapods-stats           | 1.0.0        |
| netrc                     | 0.11.0       |
| cocoapods-trunk           | 1.3.1        |
| cocoapods-try             | 1.1.0        |
| colored2                  | 3.1.2        |
| escape                    | 0.0.4        |
| fourflusher               | 2.0.1        |
| gh_inspector              | 1.1.3        |
| molinillo                 | 0.6.6        |
| ruby-macho                | 1.3.1        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.7.0        |
| cocoapods                 | 1.5.3        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| digest-crc                | 0.4.1        |
| unf_ext                   | 0.0.7.5      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20180417 |
| dotenv                    | 2.7.1        |
| emoji_regex               | 1.0.1        |
| excon                     | 0.62.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.15.4       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.13.1       |
| fastimage                 | 2.1.5        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.0       |
| multi_json                | 1.13.1       |
| os                        | 1.0.0        |
| signet                    | 0.11.0       |
| googleauth                | 0.6.7        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2018.0812  |
| mime-types                | 3.2.2        |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| google-api-client         | 0.23.9       |
| google-cloud-env          | 1.0.5        |
| google-cloud-core         | 1.3.0        |
| google-cloud-storage      | 1.16.0       |
| json                      | 2.2.0        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.5.0        |
| rubyzip                   | 1.2.2        |
| security                  | 0.1.3        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.5        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.4.1        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.6.5        |
| tty-cursor                | 0.6.1        |
| tty-spinner               | 0.9.0        |
| word_wrap                 | 1.0.0        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| xcpretty-travis-formatter | 1.0.0        |
| fastlane-plugin-ruby      | 0.1.3        |
| jaro_winkler              | 1.5.2        |
| mini_portile2             | 2.3.0        |
| nokogiri                  | 1.8.5        |
| parallel                  | 1.13.0       |
| parser                    | 2.6.0.0      |
| powerpack                 | 0.1.2        |
| rainbow                   | 3.0.0        |
| ruby-progressbar          | 1.10.0       |
| rubocop                   | 0.63.0       |
| slather                   | 2.4.6        |
</details>

*generated on:* **2019-03-08**
</details></pre>
</details>

**Comment by mollyIV** [comment]
👋`fastlane` uses [`xcpretty`](https://github.com/xcpretty/xcpretty) under the hood when it comes to generating the report for the test results. The issue is most probably related to https://github.com/xcpretty/xcpretty/issues/344. As a workaround you can try to use [trainer](https://github.com/xcpretty/trainer) 👍

**Comment by dcordero** [comment]
Still failing on 2.120.0

**Comment by dk53** [comment]
Having the exact same issue, still failing on `2.123.0` and [xcpretty](https://github.com/xcpretty/xcpretty/issues/344) doesn't look still maintained 🤷‍♂️
